runner/src/indexer/indexer.ts

```diff
@@ -219,83 +222,87 @@ export default class Indexer {
     };
   }
 
-  validateTableNames (tableNames: string[]): void {
-    if (!(Array.isArray(tableNames) && tableNames.length > 0)) {
+  getTableNames (schema: string): string[] {
+    const tableNames = this.deps.parser.tableList(schema, { database: 'Postgresql' })
+      .filter(table => table.startsWith('create::null::'))
+      .map(table => table.replace('create::null::', ''));
+    if (tableNames.length === 0) {
       throw new Error('Schema does not have any tables. There should be at least one table.');
     }
-    const correctTableNameFormat = /^[a-zA-Z_][a-zA-Z0-9_]*$/;
-
-    tableNames.forEach(name => {
-      if (!name.includes('"') && !correctTableNameFormat.test(name)) { // Only test if table name doesn't have quotes
-        throw new Error(`Table name ${name} is not formatted correctly. Table names must not start with a number and only contain alphanumerics or underscores.`);
-      }
-    });
-  }
-
-  getTableNames (schema: string): string[] {
-    const tableRegex = /CREATE TABLE\s+(?:IF NOT EXISTS\s+)?"?(.+?)"?\s*\(/g;
-    const tableNames = Array.from(schema.matchAll(tableRegex), match => {
-      let tableName;
-      if (match[1].includes('.')) { // If expression after create has schemaName.tableName, return only tableName
-        tableName = match[1].split('.')[1];
-        tableName = tableName.startsWith('"') ? tableName.substring(1) : tableName;
-      } else {
-        tableName = match[1];
-      }
-      return /^\w+$/.test(tableName) ? tableName : `"${tableName}"`; // If table name has special characters, it must be inside double quotes
-    });
-    this.validateTableNames(tableNames);
     console.log('Retrieved the following table names from schema: ', tableNames);
     return tableNames;
   }
 
-  sanitizeTableName (tableName: string): string {
-    tableName = tableName.startsWith('"') && tableName.endsWith('"') ? tableName.substring(1, tableName.length - 1) : tableName;
-    return tableName.replace(/[^a-zA-Z0-9_]/g, '_');
+  sanitizeTableName (tableName: string, tableNameCount: Map<string, number>): string {
+    // Replace special characters with underscores
+    let sanitizedName: string = tableName.replace(/[^a-zA-Z0-9_]/g, '_');
+    // Convert to CamelCase
+    sanitizedName = sanitizedName
+      .replace(/_([a-zA-Z0-9])/g, (match: string) => match.toUpperCase())
```

Comment:
```suggestion
      .replace(/_([a-zA-Z])/g, (match: string) => match.toUpperCase())
```
We shouldn't need to match numbers here?

Reply:
True. We don't need to. I'll remove it!